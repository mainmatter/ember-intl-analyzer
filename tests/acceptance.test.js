import { it } from 'vitest';
import { describe } from 'vitest';
import { execa } from 'execa';
import { expect } from 'vitest';
import { readdirSync } from 'node:fs';

describe('basic acceptance test', () => {
  it.for(readdirSync('./tests/acceptance-fixtures'))(
    'can run the cli with %s',
    async folderName => {
      const { stdout } = await execa({
        cwd: `./tests/acceptance-fixtures/${folderName}`,
      })`node ../../../bin/cli.js`;
      expect(stdout).toMatchSnapshot();
    }
  );
});
