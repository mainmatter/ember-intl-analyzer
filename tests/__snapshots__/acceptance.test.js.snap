// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic acceptance test > can run the cli with config-file 1`] = `
"[1/4] 🔍  Finding JS and HBS files...
[2/4] 🔍  Searching for translations keys in JS and HBS files...
[3/4] ⚙️   Checking for unused translations...
[4/4] ⚙️   Checking for missing translations...

 👏  No unused translations were found!

 👏  No missing translations were found!

 ⚠️   Found 1 unused whitelist entry! Please remove it from the whitelist:

   - /some\\.unused\\.whitelisted\\.translation-(a|b)/"
`;

exports[`basic acceptance test > can run the cli with config-file-cjs 1`] = `
"[1/4] 🔍  Finding JS and HBS files...
[2/4] 🔍  Searching for translations keys in JS and HBS files...
[3/4] ⚙️   Checking for unused translations...
[4/4] ⚙️   Checking for missing translations...

 👏  No unused translations were found!

 👏  No missing translations were found!

 ⚠️   Found 1 unused whitelist entry! Please remove it from the whitelist:

   - /some\\.unused\\.whitelisted\\.translation-(a|b)/"
`;

exports[`basic acceptance test > can run the cli with config-file-mjs 1`] = `
"[1/4] 🔍  Finding JS and HBS files...
[2/4] 🔍  Searching for translations keys in JS and HBS files...
[3/4] ⚙️   Checking for unused translations...
[4/4] ⚙️   Checking for missing translations...

 👏  No unused translations were found!

 👏  No missing translations were found!

 ⚠️   Found 1 unused whitelist entry! Please remove it from the whitelist:

   - /some\\.unused\\.whitelisted\\.translation-(a|b)/"
`;

exports[`basic acceptance test > can run the cli with config-file-type-module 1`] = `
"[1/4] 🔍  Finding JS and HBS files...
[2/4] 🔍  Searching for translations keys in JS and HBS files...
[3/4] ⚙️   Checking for unused translations...
[4/4] ⚙️   Checking for missing translations...

 👏  No unused translations were found!

 👏  No missing translations were found!

 ⚠️   Found 1 unused whitelist entry! Please remove it from the whitelist:

   - /some\\.unused\\.whitelisted\\.translation-(a|b)/"
`;
